### ISSUE TITLE
How to use options asynchronous?

### ISSUE THREAD

**Issue by killua99** [formly-js/vue-formly#33]
Hi,

I'm dealing with some async data for a select. Before I was calling the form like a component was working just fine because I pass a props to the component form with the data resolved, now, I move it into a router address, and now the form is the component itself.

This is how looks my code to bring the options

```js
        {
          key: 'fk_artist',
          type: 'select',
          required: true,
          options: this.artistList,
          templateOptions: {
            label: this.$tc('artworks.form.fk_artist'),
            classes: {
              'form__select': true
            }
          }
        },
```
Now since the created () function is way to late and the data is asynchronous the form render without the options. There is a way to wait for the options or to pause the form render until all my async calls are done?

**Comment by matt-sanders** [comment]
Hmm, you may have to use watchers or computed properties to do this. Maybe have a 'loaded' flag that conditionally displays the form. When your data is loaded, populate the fields and then switch the loaded flag on or something. Because all the options are returned from `data()` I don't think you'll be able to connect responsive data to them.

**Comment by matt-sanders** [comment]
or, alternatively, you could have all your fields as a computed property, which will respond to async updates.

**Comment by killua99** [comment]
What I did so far and is working.

I did a mixin with my fields, form, model, so good so far nothing speacil here, also I did a declaration with artistList: [],

On my form component, I call my mixins, and at created () I did all the fetch data and fix my options, return my option inside `artistList` so, when the created is done the options is filled with the data I wanted. But it wasn't enough. So I populate my fields with the options with something like this.

```js
      _.each(this.fields, field => {
        if (field.key === 'fk_artist') field.options = this.artistList
        if (field.key === 'fk_warehouse') field.options = this.warehouseList
      })
```

Without that last lines, the options just went empty, no update at all. I'm wondering if the options are reactive, I'll keep playing to know if I change option during the form is already render if options changes too.

I don't know if is the best option, but the form saves, and the model seems to have the right values when I update data.
